packages/cli/src/oclif/commands/promote.js

```diff
@@ -43,6 +43,50 @@ const hasAppChangeType = (metadata, changeType) => {
 };
 
 class PromoteCommand extends BaseCommand {
+  async run_require_confirmation_pre_checks(app, requestBody) {
```

Comment:
hmmm this seems duplicative of [`run_require_confirmation_pre_checks`](https://github.com/zapier/zapier-platform/pull/951/files#diff-896ab13a96dca05c7bf8e2db52daaab026388f54621cf0d9efb24d714fb09a0dR10) in `MigrateCommand`, would it make sense to pull it into a shared resource?

Reply:
I thought about it, but then we'd also have to pass in the `jobType` for example in order to separate logs for `promote` and `migrate`, and refactor the logic further in order to accommodate the shared resource (for example, we could create a subclass for `ZapierBaseCommand` since these methods are a bit too specific to migrations for that base class.)

Personally, I believe that two similar methods is OK; three is where a refactor should be considered 😅

Reply:
I am okay with leaving it as is 👍 !